### ISSUE TITLE
[BUG] docker compose watch needs sudo (DD4W WSL2)

### ISSUE THREAD

**Issue by ktzsolt** [docker/compose#11334]
### Description

I was trying to use the new `docker compose watch` and it works only if I explicitly add `sudo` in front of the command, however my user is in the docker group and all other docker commands work without sudo. 

without sudo I get this error: 

> cannot take exclusive lock for project "composewatch": open /run/user/1000/composewatch.pid: no such file or directory

I issue docker commands from a WSL2 ubuntu distro towards the DD4W docker engine using the DD4W WSL2 integration feature.

### Steps To Reproduce

1. On Windows 10 with Docker Desktop for Windows (4.26.1 (131620) ) inside a ubuntu:22.04 WSL2 distro (that is enabled in DD settings WSl2 integration) add your user to the docker group.
2. Issue `docker compose watch` command from inside a ubuntu:22.04 WSL2 distro without sudo in this example repository:  https://github.com/ktzsolt/composewatch

### Compose Version

```Text
Docker Compose version v2.23.3-desktop.2
```

### Docker Environment

```Text
Client: Docker Engine - Community
 Version:    24.0.6
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.0-desktop.2
    Path:     /usr/local/lib/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.23.3-desktop.2
    Path:     /usr/local/lib/docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.21
    Path:     /usr/local/lib/docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  0.1
    Path:     /usr/local/lib/docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.10
    Path:     /usr/local/lib/docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-scan
  scout: Docker Scout (Docker Inc.)
    Version:  v1.2.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-scout

Server:
 Containers: 6
  Running: 4
  Paused: 0
  Stopped: 2
 Images: 35
 Server Version: 24.0.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d8f198a4ed8892c764191ef7b3b06d8a2eeb5c7f
 runc version: v1.1.10-0-g18a0cb0
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.133.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 13.66GiB
 Name: docker-desktop
 ID: 2d16933a-1c6b-4b61-9269-f1eaca97abcf
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
```

**Comment by ktzsolt** [comment]
Yup, I have. owner and group of the directory /run/user/1000 (1000, that is myuser's uid) is myuser:myuser

**Comment by ktzsolt** [comment]
Also I have systemd enabled in the ubuntu distro, my `/etc/wsl.conf` file is like this:
```
[interop]
appendWindowsPath = false
[boot]
systemd=true
```

**Comment by ktzsolt** [comment]
It looks like that /run/user/1000 directory was created by me by hand before when I was trying to fix this, I realized this by checking my cmd HISTORY. 

But after restart the /run/user/1000 directory didn't get created so I issued an `sudo apt-get upgrade` command and it looks like it fixed the problem, now `docker compose watch` runs without sudo and `/run/user/1000/composewatch.pid` gets created. Maybe this was some systemd problem.

Thanks for putting me in the right direction!
